signer/src/transaction_signer.rs

```diff
@@ -738,6 +749,36 @@ where
         client.can_accept(address).await.unwrap_or(false)
     }
 
+    async fn can_accept_deposit_request(&self, req: &model::DepositRequest) -> Result<bool, Error> {
+        // If we have not configured a blocklist checker, then we can
+        // return early.
+        let Some(client) = self.blocklist_checker.as_ref() else {
+            return Ok(true);
+        };
+
+        // We turn all the input scriptPubKeys into addresses and check
+        // those with the blocklist client.
+        let bitcoin_network = bitcoin::Network::from(self.context.config().signer.network);
+        let params = bitcoin_network.params();
+        let addresses = req
```

Comment:
Question: This is the stuff that crashed before, right? When working towards the demo and trying to convert to an address the blocklist client would accept? Do we know if this code fixes that?

Reply:
> This is the stuff that crashed before, right? When working towards the demo and trying to convert to an address the blocklist client would accept?

Yeah, the code would crashed here.

> Do we know if this code fixes that if so?

That bug was fixed by https://github.com/stacks-network/sbtc/pull/675. This change is just a reorganization.

Reply:
Awesome, just wanted to double check👌